### ISSUE TITLE
Improve history overview

### ISSUE THREAD

**Issue by weihwang** [jupyterlab/jupyterlab-git#245]
Proposal for full width tabbed history view, which is related to how we may think about push/pull actions. We need a more detailed higher level view of where the latest changes are in local/remote branches. Fly-out full detail on mouse-over provides an easier "browse" experience. Adding a link at the bottom of a default 10 item list gives users a link to the full log if needed.

https://www.figma.com/file/UyNpeQNuTj6Uv74MoEJDh3ji/Git-extension?node-id=517%3A1718

![image](https://user-images.githubusercontent.com/551333/46969738-dfd29c80-d06b-11e8-9451-4a9dc7a7d33f.png)

**Comment by dhirschfeld** [comment]
I was thinking that it might be very cool you you could select two nodes in the history to view the diff between them.

**Comment by tgeorgeux** [comment]
I put a similar response in #244, what is the upside of fly-out versus just expanding/collapsing each history field?

**Comment by tgeorgeux** [comment]
I definately agree with the onclick. I think the fly-out as an overview has
merit, but the interactions should stay within the sidebar.

@dhirschfeld , I think that diff idea is a great idea! Can you open a
separate issue with the 'enhancement' tag, so we can track where this is at
and will be in the existing milestones without losing that suggestion?

**Comment by jaipreet-s** [comment]
cc @saulshanabrook

**Comment by saulshanabrook** [comment]
I would be in favor of the on click over the popout on hover (I find hovering sorta fiddly often).

But am happy to get started on either approach if we consensus on which to pursue.

**Comment by tgeorgeux** [comment]
@weihwang @jaipreet-s Do you have additional thoughts on popout vs on-click expanding?

**Comment by weihwang** [comment]
If it's not possible today to create a popout, and the general feedback leans towards inline expansion of details, I think we can start with that version. I have a mock below to expand-on-click. I needed to add more details (usesrname, commit hash) into the collapsed view in this case to create a more seamless "expand" to minimize disrupting content when going to the detail state.

**Comment by jaipreet-s** [comment]
Inline expansion looks better to me as well.

In general, I'd suggest taking this one incrementally

1. Get the "summary view" in place (This'll also unblock the push/pull task)
2. Add "detailed view" on inline expansion
3. Add extra details like the `View mode in Terminal option`

**Comment by tgeorgeux** [comment]
Sounds like a good plan to me.

**Comment by dhirschfeld** [comment]
`master/origin` strikes me as back-to-front - wouldn't it be more usual to refer to it as `remote/branch`?

Referring to it in the opposite way to what the cli does might make it more difficult for new users to pick up the command line syntax.

**Comment by weihwang** [comment]
@dhirschfeld I agree we should align with the expected display. I may have misrepresented the order of the remote/branch. I'll update the mockup. Thanks!
